src/emulator/gxm/src/gxm.cpp

```diff
@@ -42,31 +42,47 @@ static const char *parameter_name(const SceGxmProgramParameter &parameter) {
     return reinterpret_cast<const char *>(bytes + parameter.name_offset);
 }
 
-static const char *scalar_type(SceGxmParameterType type) {
+static SceGxmParameterType parameter_type(const SceGxmProgramParameter &parameter) {
+    return static_cast<SceGxmParameterType>(
+        static_cast<uint16_t>(parameter.type));
+}
+
+static const char *scalar_type(SceGxmParameterType const& type) {
```

Comment:
Missing SCE_GXM_PARAMETER_TYPE_AGGREGATE, maybe add a warn for it?

Reply:
yea I had it initially but I removed it because I added `default` case, so it warns already.